src/ranch.erl

```diff
@@ -447,6 +449,20 @@ procs1(ListenerSup, connections) ->
 	),
 	lists:flatten(Conns).
 
+-spec stats(ref()) -> #{pos_integer() => {non_neg_integer(), non_neg_integer()}}.
+stats(Ref) ->
+	Counters = ranch_server:get_stats_counters(Ref),
+	CounterInfo = counters:info(Counters),
+	NumCounters = maps:get(size, CounterInfo),
+	NumConnsSups = NumCounters div 2,
+	lists:foldl(
+		fun (Id, Acc) ->
+			Acc#{Id => {counters:get(Counters, 2*Id-1), counters:get(Counters, 2*Id)}}
+		end,
+		#{},
+		lists:seq(1, NumConnsSups)
+	).
```

Comment:
Hm I don't think the stats format works. The stats should follow a format where adding new values doesn't change existing values.

So I would suggest something like this:

``` erlang
#{
    {conns_sup, Id, accept} => non_neg_integer(),
    {conns_sup, Id, terminate} => non_neg_integer()
}
```

Then we can add as many values as we want in the future.

Reply:
Ok then, I will change it as you suggest.

Reply:
done